File: app/src/main/java/com/tanish2k09/sce/fragments/containerFragments/CategoryFragment.java
package com.tanish2k09.sce.fragments.containerFragments;

import android.content.Context;
import android.content.SharedPreferences;
import android.graphics.Color;
import android.os.Build;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.cardview.widget.CardView;
import androidx.fragment.app.Fragment;

import com.tanish2k09.sce.R;

import java.util.Objects;

import static android.content.Context.MODE_PRIVATE;

/**
 * A simple {@link Fragment} subclass.
 * create an instance of this fragment.
 */
public class CategoryFragment extends Fragment {

    private String categoryName = "Not categorized";
    private CardView catCard;

    public CategoryFragment() {
        // Required empty public constructor
    }

    public void setInstanceCategory(String category) {
        categoryName = category;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View v = inflater.inflate(R.layout.fragment_category, container, false);

        TextView catText = v.findViewById(R.id.categoryTitle);
        catText.setText(categoryName);

        catCard = v.findViewById(R.id.catCard);

        return v;
    }

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);
    }

    @Override
    public void onResume() {
        super.onResume();
        SharedPreferences sp = catCard.getContext().getSharedPreferences("settings",MODE_PRIVATE);
        int accent = Color.parseColor(sp.getString("accentCol", "#00bfa5"));
        catCard.setCardBackgroundColor(accent);

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
            catCard.setOutlineAmbientShadowColor(accent);
            catCard.setOutlineSpotShadowColor(accent);
        }
    }

    public void pushConfigVarFragment(fConfigVar var, int index) {
        getChildFragmentManager().beginTransaction().add(R.id.categoryList, var, "configCard"+index).commit();
    }

    public boolean popFragment(int index) {
        getChildFragmentManager().beginTransaction()
                                    .remove(Objects.requireNonNull(getChildFragmentManager()
                                        .findFragmentByTag("configCard" + index)))
                                    .commitNow();
        return getChildFragmentManager().getFragments().size() == 0;
    }
}

File: app/src/main/java/com/tanish2k09/sce/fragments/containerFragments/CategoryFragment.kt
package com.tanish2k09.sce.fragments.containerFragments

import android.graphics.Color
import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

import com.tanish2k09.sce.R

import android.content.Context.MODE_PRIVATE

/**
 * A simple [Fragment] subclass.
 * create an instance of this fragment.
 */
class CategoryFragment : Fragment() {

    private var categoryName = "Not categorized"
    private var dividerTop: View? = null
    private var dividerBottom: View? = null
    private var catText: TextView? = null

    fun setInstanceCategory(category: String) {
        categoryName = category
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        // Inflate the layout for this fragment
        val v = inflater.inflate(R.layout.fragment_category, container, false)

        catText = v.findViewById(R.id.categoryTitle)
        catText!!.text = categoryName

        dividerTop = v.findViewById(R.id.catDivTop)
        dividerBottom = v.findViewById(R.id.catDivBottom)

        return v
    }

    override fun onResume() {
        super.onResume()
        val sp = catText!!.context.getSharedPreferences("settings", MODE_PRIVATE)
        val accent = Color.parseColor(sp.getString("accentCol", "#00bfa5"))
        catText!!.setTextColor(accent)
        dividerBottom!!.setBackgroundColor(accent)
        dividerTop!!.setBackgroundColor(accent)
    }

    fun pushConfigVarFragment(`var`: ConfigVar, index: Int) {
        childFragmentManager.beginTransaction().add(R.id.categoryList, `var`, "configCard$index").commit()
    }

    fun popFragment(index: Int): Boolean {
        childFragmentManager.beginTransaction()
                .remove(childFragmentManager.findFragmentByTag("configCard$index")!!)
                .commitNow()
        return childFragmentManager.fragments.size == 0
    }
}// Required empty public constructor
